### ISSUE TITLE
Part of title bar is still rendered when it should not

### ISSUE THREAD

**Issue by yagudron** [lepoco/wpfui#595]
### Describe the bug

I recently started using the library, and whilst working on Windows 10 machine everything looked as expected. When I moved my work to Windows 11, I noticed some part of the title bar is rendered, even when shouldn't. Attaching examples of that reproduced for the WPF UI Gallery. I have not yet found any workarounds to remove it in my app too.

### To Reproduce

Start WPF UI Gallery or demo app.

### Expected behavior

No additional elements in the title bar.

### Screenshots

Windows 10:
![Win10](https://user-images.githubusercontent.com/7910223/232444483-8cd05dac-3da1-483a-bc60-ff436d130422.jpg)

Windows 11:
![Win11](https://user-images.githubusercontent.com/7910223/232444558-638fb6ed-1ae7-48a9-b344-e3ac81074e4c.jpg)

### OS version

Edition	Windows 11 Home
Version	22H2
Installed on	‎22/‎10/‎13
OS build	22621.1555
Experience	Windows Feature Experience Pack 1000.22640.1000.0

### .NET version

7.0

### WPF-UI NuGet version

2.0.3

**Comment by IvanDmitriev1** [comment]
Take a look at the development branch, does the problem persist?

**Comment by yagudron** [comment]
Yes, I have tried building from `development` - same behavior.

![image](https://user-images.githubusercontent.com/7910223/232467015-6f7b83c3-2e4e-489a-82f0-bd24dca99ed4.png)

**Comment by yagudron** [comment]
I have found the Windows setting which causes this.

Personalization > Colors
![image](https://user-images.githubusercontent.com/7910223/232479573-8d0235cc-e7d9-456f-b583-c202a4e1f50c.png)

It is reproduced only when it's on.

**Comment by IvanDmitriev1** [comment]
Strangely, after enabling this option the title bar has not changed. But for example google chrome has changed. 
![image](https://user-images.githubusercontent.com/42055372/232517065-c3e8ae08-d8e0-4efd-9de6-cd18f8941f03.png)

**Comment by yagudron** [comment]
Interesting. I even asked my friend to try and he got the same results. Maybe there's more to it then.

**Comment by olabacker** [comment]
It only seems to occur when the window is focused. Anyone has a workaround yet?

**Comment by syntax-tm** [comment]
## The Problem

This is caused by the user's Accent color being applied to the Window's titlebar.

## Solutions

The quick workaround is to not have `Show accent color on title bars and windows borders` checked in the Windows Settings under `Personalization > Colors`.

If you don't want to change the setting via the Settings app, you can either do this directly in the Registry using `regedit` or via Powershell.

**Using `regedit`:**

Open `regedit` and navigate to `HKCU:\Software\Microsoft\Windows\DWM` for an item named `ColorPrevalence`. If you don't see this then right-click and create a new `DWORD` with that name. Set the value to `0` in order to disable it or `1` to enable it.

**Using `powershell`:**

```ps1
# check current ColorPrevalence setting
Get-ItemPropertyValue -Path 'HKCU:\Software\Microsoft\Windows\DWM' -Name 'ColorPrevalence'

# disable ColorPrevalence (Accent colors on window titlebars and borders)
Set-ItemProperty -Path 'HKCU:\Software\Microsoft\Windows\DWM\' -Name 'ColorPrevalence' -Type DWord -Value 0
```

### Other Methods

Aside from `regedit` and `powershell` you can create a `.reg` file, use the `reg` command in `cmd`, set via Group Policy, etc.

## (Actually) Solving the Problem

Other than turning that Personalization setting off, there's likely a native Win32 API call that can be used to disable the title bar coloring with DWM. If I have some extra free time I'll see if I can put something together but for now here's a link to the documentation from Microsoft.

- [Desktop Window Manager (DWM) | Microsoft Docs](https://learn.microsoft.com/en-us/windows/win32/api/_dwm/)

**Comment by chenxuuu** [comment]
![图片](https://github.com/lepoco/wpfui/assets/10357394/d28a64a4-f029-4a0b-9a3a-bb3763024b59)

still have this problem

**Comment by Jay-o-Way** [comment]
bump

**Comment by Jay-o-Way** [comment]
Isn't it possible to set a window style that removes the titlebar, for a quick and dirty work-around?

**Comment by drolevar** [comment]
Workaround:

Put this into the MainWindow constructor:

```
    public MainWindow()
    {
        InitializeComponent();

        new WindowInteropHelper(this).EnsureHandle();

        SystemThemeWatcher.Watch(this, WindowBackdropType.Auto);
    }
```

The idea is that the window handle should be created before `SystemThemeWatcher.Watch(...)` runs.

This is because underneath there is a call to `WindowBackdrop.RemoveBackdrop(...)` and that fails if the window handle is not created.

In general I think that the whole idea of "lazy" window initialization is flawed.

**Comment by 537mfb** [comment]
The actual issue is that `ui:titlebar` is taller then the area that get's colored by the accent color and the title is set to beat the center of the `ui:titlebar`

if you give the `ui:titlebar` a height of 30 this issue goes away

`<ui:TitleBar Height="30" ... />`

Before adding an height of 30
![image](https://github.com/lepoco/wpfui/assets/1590767/e0cb2f4b-8473-4981-806a-6671021988b1)

After adding an height of 30
![image](https://github.com/lepoco/wpfui/assets/1590767/9c22b7e3-0403-4394-a001-53d46da521df)

The real issue with titlebar is that the icons on the system buttons stay black on dark mode until you hover the mouse on them

At start
![image](https://github.com/lepoco/wpfui/assets/1590767/ee19eb8e-846d-4e6a-a6ab-37384f07ab2b)

after hovering over the minimize button
![image](https://github.com/lepoco/wpfui/assets/1590767/57fdad3d-b778-4f0d-9c42-69bf10973ecd)

After hovering over all system buttons
![image](https://github.com/lepoco/wpfui/assets/1590767/59aafd25-3c78-49ae-800b-e2dffa2877d8)

for this issue I haven't yet figured out a solution

**Comment by Jay-o-Way** [comment]
> if you give the `ui:titlebar` a height of 30 this issue goes away

Should be 32, but #details :)

**Comment by 537mfb** [comment]
Actually at least on my PC the height of the accent colored bar os 30px exactly

How did you get 32px @Jay-o-Way?

**Comment by Jay-o-Way** [comment]
> How did you get 32px @Jay-o-Way?

Well, I mean, according to the design... https://learn.microsoft.com/windows/apps/design/basics/titlebar-design#standard-design

**Comment by 537mfb** [comment]
> Well, I mean, according to the design...

Interesting - an old wpf app without wpf-ui actually has 31px of accented colored titlebar while in my wpf-ui test app it's only 30px

Perhaps that 32px includes a border?

**Comment by aquinn39** [comment]
The fix for this issue is to use DwmSetWindowAttribute to set the DWMWA_CAPTION_COLOR attribute to 0xFFFFFFFE - this causes the title bar colour to not show when the mica effect is enabled with DWMWA_SYSTEMBACKDROP_TYPE (I'm assuming this is what wpfui uses to enable the mica effect.). However, this behaviour is undocumented [here](https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/ne-dwmapi-dwmwindowattribute).

**Comment by pomianowski** [comment]
Fixed`?` in #1122
